pkg/controllers/managementagent/endpoints/endpoints.go

```diff
@@ -35,87 +30,38 @@ const (
 	endpointsAnnotation = "field.cattle.io/publicEndpoints"
 )
 
-func Register(ctx context.Context, workload *config.UserContext) {
-	isRKE := false
-	cluster, err := workload.Management.Management.Clusters("").Get(workload.ClusterName, metav1.GetOptions{})
-	if err != nil {
-		logrus.WithError(err).Warnf("Can not get cluster %s when registering endpoint controller", workload.ClusterName)
-	}
-	if cluster != nil {
-		//assume that cluster always has a spec
-		isRKE = cluster.Spec.RancherKubernetesEngineConfig != nil
-	}
-
-	ignore := func() bool {
-		return false
-	}
-	if cluster.Spec.Internal {
-		deployments := workload.Apps.Deployments("").Controller().Lister()
-		ignore = func() bool {
-			_, err := deployments.Get(namespace.System, "cattle-cluster-agent")
-			return err == nil
-		}
-	}
```

Comment:
I don't see any equivalent to this, should we still skip it for local clusters?

Reply:
The logic before was if the user controller was running for the local cluster and rancher is managing this cluster, then don't run.  This is the hosted rancher setup.  Now that we are running the agent we don't need to do this because the agent will not run controllers if the local cluster has rancher installed.  So it's not possible to get two instances of rancher both trying to manage the endpoints